pkg/resource/node_resource_manager.go

```diff
@@ -177,23 +188,91 @@ func (o *NodeResourceManager) FindTargetNode(tsp *predictionapi.TimeSeriesPredic
 	return false, nil
 }
 
-func (o *NodeResourceManager) BuildNodeStatus(tsp *predictionapi.TimeSeriesPrediction, node *v1.Node) {
-	idToResourceMap := map[string]*v1.ResourceName{}
-	for _, metrics := range tsp.Spec.PredictionMetrics {
-		if metrics.ResourceQuery == nil {
+func (o *NodeResourceManager) BuildNodeStatus(node *v1.Node) map[v1.ResourceName]string{
+	tspCanNotBeReclaimedResource := o.GetCanNotBeReclaimedResourceFromTsp(node)
+	localCanNotBeReclaimedResource := o.GetCanNotBeReclaimedResourceFromLocal()
+	reserveCpuPercent := o.reserveResource.CpuPercent
+	if nodeReserveCpuPercent, ok := getReserveResourcePercentFromNodeAnnotations(node.GetAnnotations(), v1.ResourceCPU.String()); ok {
+		reserveCpuPercent = &nodeReserveCpuPercent
+	}
+
+	extResourceFrom := map[v1.ResourceName]string{}
+
+	for resourceName, value := range tspCanNotBeReclaimedResource {
+		resourceFrom := "tsp"
+		maxUsage := value
+		if localCanNotBeReclaimedResource[resourceName] > maxUsage {
+			maxUsage = localCanNotBeReclaimedResource[resourceName]
+			resourceFrom = "local"
+		}
+		var nextRecommendation float64
+		switch resourceName {
+		case v1.ResourceCPU:
+			// cpu need to be scaled to m as ext resource cannot be decimal
+			if reserveCpuPercent != nil {
+				nextRecommendation = (float64(node.Status.Allocatable.Cpu().Value())**reserveCpuPercent - maxUsage) * 1000
```

Comment:
why here used ** ?

Reply:
Because the riseCpuPercent is a pointer, I need to use the pointer to determine whether the reserveCpuPercent is set